src/main/kotlin/no/nav/k9/ettersending/EttersendingService.kt

```diff
@@ -1,73 +1,63 @@
-    ){
-        logger.info("Registrerer ettersending av typen ${ettersending.søknadstype.name}. Henter søker")
-        val søker: Søker = søkerService.getSoker(idToken = idToken, callId = callId)
+        callId: CallId,
+        metadata: Metadata
+    ) {
+        logger.info(formaterStatuslogging(ettersending.søknadId, "registreres"))
+        ettersending.valider()
 
-        logger.trace("Søker hentet. Validerer søker.")
+        val søker: Søker = søkerService.getSoker(idToken = idToken, callId = callId)
         søker.validate()
-        logger.trace("Søker Validert.")
-
-        logger.info("Henter ${ettersending.vedlegg.size} vedlegg.")
-        val vedlegg = vedleggService.hentVedlegg(
-            idToken = idToken,
-            vedleggUrls = ettersending.vedlegg,
-            callId = callId,
-            eier = DokumentEier(søker.fødselsnummer)
-        )
-
-        logger.trace("Vedlegg hentet. Validerer vedlegg.")
-        vedlegg.validerVedlegg(ettersending.vedlegg)
-        logger.info("Vedlegg validert")
 
-        logger.info("Legger ettersending til prosessering")
-
-        val komplettEttersending = KomplettEttersending(
-            søker = søker,
-            språk = ettersending.språk,
-            mottatt = ZonedDateTime.now(ZoneOffset.UTC),
-            vedlegg = vedlegg,
-            søknadId = ettersending.søknadId,
-            harForståttRettigheterOgPlikter = ettersending.harForståttRettigheterOgPlikter,
-            harBekreftetOpplysninger = ettersending.harBekreftetOpplysninger,
-            beskrivelse = ettersending.beskrivelse,
-            søknadstype = ettersending.søknadstype,
-            k9Format = k9Format
-        )
-
-        k9EttersendingMottakGateway.leggTilProsesseringEttersending(
-            ettersending = komplettEttersending,
-            callId = callId
-        )
-
-        logger.trace("Ettersending lagt til mottak.")
+        logger.info("Validerer ${ettersending.vedlegg.size} vedlegg")
+        val dokumentEier = DokumentEier(søker.fødselsnummer)
+        val vedleggHentet = vedleggService.hentVedlegg(ettersending.vedlegg, idToken, callId, DokumentEier(søker.fødselsnummer))
+        vedleggHentet.validerVedlegg(ettersending.vedlegg)
+
+        logger.info("Persisterer vedlegg")
+        vedleggService.persisterVedlegg(ettersending.vedlegg, callId, dokumentEier)
+
+        val titler = vedleggHentet.map { it.title }
+        val mottatt = ZonedDateTime.now(ZoneOffset.UTC)
+        val k9Format = ettersending.tilK9Format(mottatt, søker)
+        val komplettEttersending = ettersending.tilKomplettEttersending(k9Format, søker, k9MellomLagringIngress, mottatt, titler)
+
+        try {
+            kafkaProducer.produserKafkaMelding(komplettEttersending, metadata)
+        } catch (exception: Exception) {
+            logger.info("Feilet ved å legge melding på Kafka. Sletter persisterte vedlegg")
+            vedleggService.slettPersistertVedlegg(ettersending.vedlegg, callId, dokumentEier)
```

Comment:
Sletter denne kun persisteringen eller hele vedlegget?

Reply:
Hm godt spørsmål

Reply:
Kan se for meg scenario der søker sender med lenke til et vedlegg. Appen feiler med å legge melding på kafka, og sletter vedlegget. Søker klikker seg tilbake og (potensielt) prøver på nytt. Denne gangen har vi referanse til et vedlegg som ikke lenger eksisterer, og vil da gi en valideringsfeil til bruker om at vi har fått referanse til et vedlegg som ikke lenger eksisterer.

Reply:
Ja det er jo et reelt scenario. Kjapp titt i kildekoden til k9-mellomlagring så virker det som delete faktisk sletter.

Reply:
Ok. Godt mulig vi må da utvide apiet på k9-mellomlagring til å kunne fjernet hold (persistering) på dokumentet, med mindre det allerede eksisterer en?

Reply:
Ja det burde jo være greit å legge til. Nei er bare et delete endepunkt nå og den fjerner alt ser det ut som.

Reply:
https://github.com/navikt/k9-mellomlagring/blob/master/src/main/kotlin/no/nav/helse/dokument/storage/GcpStorageBucket.kt#L39